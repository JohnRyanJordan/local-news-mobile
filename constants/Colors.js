export default {
  base: '#fff',
  baseText: '#000',
  overlayTextBackground: '#1119',
  tabIconDefault: '#ccc',
  tabIconSelected: '#2f95dc',
  tabSelectedUnderline: '#A32921',
  tabBar: '#fefefe',
  errorBackground: 'red',
  errorText: '#fff',
  warningBackground: '#EAEB5E',
  warningText: '#666804',
  noticeBackground: '#2f95dc',
  noticeText: '#fff',
};
