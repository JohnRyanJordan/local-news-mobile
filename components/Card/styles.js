import { StyleSheet } from 'react-native';
import Colors from '../../constants/Colors';

export default StyleSheet.create({
  container: {
    flex: 1,
    height: 200,
    margin: 5,
    backgroundColor: Colors.base,
  },
  thumbnail: {
    flex: 1,
    resizeMode: "cover",
    justifyContent: 'flex-start',
  },
  header: {
    width: '100%',
    backgroundColor: Colors.overlayTextBackground,
    justifyContent: 'center',
    alignItems: 'center',
    flexShrink: 1,
  },
  titleText: {
    textAlign: 'center',
    flexWrap: 'wrap',
    fontSize: 14,
    color: Colors.base,
  },
});