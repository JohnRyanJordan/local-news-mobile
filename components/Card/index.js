import * as React from 'react';
import { StyleSheet, Text, ImageBackground, View } from 'react-native';
import styles from './styles';

export function Card(props) {
  return (
    <View style={styles.container} testID={'container'}>
      <ImageBackground style={styles.thumbnail} source={ __DEV__ ? require('../../assets/images/robot-dev.png') : { uri: props.article.image }} testID={'backgroundImage'} >
        <View style={styles.header}>
          <Text style={styles.titleText} testID={'titleText'}>{props.article.title}</Text>
        </View>
      </ImageBackground>
    </View>
  );
}