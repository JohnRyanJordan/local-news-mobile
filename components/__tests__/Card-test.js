import * as React from 'react';
import {render, cleanup} from 'react-native-testing-library';
import { toHaveProp } from '@testing-library/jest-native';

expect.extend({ toHaveProp });
import { Card } from '../Card';
import Colors from '../../constants/Colors';

afterEach(cleanup);

describe('<Card />', () => {
  it('should match snapshot', () => {
    const rendered = render(<Card article={{title: 'Test'}} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('should properly render the title text', () => {
    const rendered = render(<Card article={{title: 'Test'}} />);
    const textComponent = rendered.getByTestId('titleText');

    expect(textComponent.props.children).toEqual('Test');
  });

  it('should properly render the image', () => {
    const rendered = render(<Card article={{title: 'Test', image: require('../../assets/images/robot-dev.png')}} />);
    const imageBackgroundComponent = rendered.getByTestId('backgroundImage');
    expect(imageBackgroundComponent).toHaveProp('source', require('../../assets/images/robot-dev.png'));
  });
});