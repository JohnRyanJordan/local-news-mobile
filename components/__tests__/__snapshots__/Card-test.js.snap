// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 200,
      "margin": 5,
    }
  }
  testID="container"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="backgroundImage"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1119",
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "flexWrap": "wrap",
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        testID="titleText"
      >
        Test
      </Text>
    </View>
  </View>
</View>
`;
